feat: support Powershell 5

PIYUSH194 opened the pull request:
### Prerequisites
- [ ] Fix iswindows  variable found in windows powershell

### Description

If iswindows variable not available in windows powershell it will through chmod command not found.
not applicable on powershell 6 and greater

AjayKMehta commented on README.md:
```diff
@@ -212,14 +212,28 @@ Feel free to use the public helper functions `Get-VCSStatus`, `Get-VcsInfo`, `Ge
 
+### Adding stack count to a custom theme
+
+As it seems getting access to the stacl information when using pushd/popd is sort of mission impossible from within a theme, you can use a workaround proposed by [Jonathan Leech-Pepin][jleechpe]. In your `$PROFILE`, add a variable that will act as a correctly scoped pointer to fetch the stack context:
```
You misspelled stack (**stacl**).

JanDeDobbeleer replied:
goddamnit, that's what you get when you're coding while watching cartoons with the kids...

PIYUSH194 committed:
Resolve IsWindows variable not found

If IsWindows variable not available in Windows PowerShell it will through chmod command not found.
Not applicable on PowerShell 6 greater

JanDeDobbeleer commented:
Wouldn't it semantically make more sense to check for PSVersion > 5?

JanDeDobbeleer commented:
Given that support for 6 isn't available, I'd opt for lowering the version number to 5 if this is all that's needed to support 5. I suppose you're using this code in PS5?

PIYUSH194 commented:
It works fine in PS5.. why no support for PS5 ?

JanDeDobbeleer commented:
That was my question indeed. Allow me to extend this properly tomorrow.

JanDeDobbeleer commented on packages/powershell/oh-my-posh/oh-my-posh.psm1:
```diff
@@ -20,7 +20,7 @@ function Get-PoshCommand {
 
 # Set the right binary to executable before doing anything else
-if (!$IsWindows) {
+if (!$IsWindows  -and (Get-Item "Variable:*IsWindows*")) {
```
Can we change this to: `if ($PSVersionTable.PSEdition -eq "Core" -and !$IsWindows)`
Which means we can change `oh-my-posh.psm1` to include 5 as well within the scope of this PR and use `feat: support Powershell 5` as a commit message.

PIYUSH194 replied:
How can i add only last commit to this pull request? Or you can close this request and add it manually

JanDeDobbeleer replied:
If you need to know how, learning to rebase is key here -> https://www.algolia.com/blog/master-git-rebase/
I can always pick this change, you stay the author in that case, let me know what would work for you.

PIYUSH194 replied:
Just pick the change that will be good.

PIYUSH194 committed:
Merge branch 'main' into main

PIYUSH194 committed:
feat: support Powershell 5
